### PR TITLE
fix compile for window.

### PR DESCRIPTION
**Pull request by baluoteliz**
fix  the issue for windows compile

### CODE REVIEW

**Comment by baluoteliz** [comment]
@yuanbo-zhang please check it.

File: CMakeLists.txt [hunk 1]
```diff
@@ -281,8 +281,22 @@ else()
         "-ldl -Wl,--whole-archive -Wl,--version-script=${CMAKE_CURRENT_SOURCE_DIR}/scripts/xquic.lds"
         ${SSL_LIB_PATH}
         "-Wl,--no-whole-archive"
-        -lpthread
-    )
+        -lpthread)
+
+    if(CMAKE_SYSTEM_NAME MATCHES "Windows")
+        target_link_libraries(xquic -lws2_32 -lgdi32 -lcrypt32)
```
Comment (yuanbo-zhang) [inline CMakeLists.txt:287]: Are you having problems when compile，need to add -lws2_32 -lgdi32 -lcrypt32，Can you describe the problem you're having

Comment (baluoteliz) [reply CMakeLists.txt:287]: yes, I have many unresolved function symbols about the socket. and others.

**Comment by Kulsk** [comment]
reconstructed and fixed windows compile problems on #383

**Pull request closed without merging**